File: test/cs/case51_inheritance.cs
using System;

public static class Program
{
    public class Base51
    {
        public int value()
        {
            return 51;
        }
    }

    public class Child51 : Base51
    {
        public int value2()
        {
            return (this.value() + 1);
        }
    }

    public static void Main(string[] args)
    {
        Child51 c = new Child51();
        Console.WriteLine(c.value2());
    }
}

File: test/py/case51_inheritance.py
class Base51:
    def value(self) -> int:
        return 51


class Child51(Base51):
    def value2(self) -> int:
        return self.value() + 1


if __name__ == "__main__":
    c: Child51 = Child51()
    print(c.value2())
